modules/generative-aws/config/class_settings.go

```diff
@@ -0,0 +1,385 @@
+		}
+		topP := ic.TopP()
+		if topP != nil && (*topP < 0 || *topP > 1) {
+			errorMessages = append(errorMessages, fmt.Sprintf("%s has to be an integer value between 0 and 1", topPProperty))
+		}
+
+		endpoint := ic.Endpoint()
+		if endpoint != "" {
+			errorMessages = append(errorMessages, fmt.Sprintf("wrong configuration: %s, not applicable to %s", endpoint, service))
+		}
+	}
+
+	if isSagemaker(service) {
+		endpoint := ic.Endpoint()
+		if endpoint == "" {
+			errorMessages = append(errorMessages, fmt.Sprintf("%s cannot be empty", endpointProperty))
+		}
+		model := ic.Model()
+		if model != "" {
+			errorMessages = append(errorMessages, fmt.Sprintf("wrong configuration: %s, not applicable to %s. did you mean %s", modelProperty, service, targetModelProperty))
+		}
+	}
+
+	if len(errorMessages) > 0 {
+		return fmt.Errorf("%s", strings.Join(errorMessages, ", "))
+	}
+
+	return nil
+}
+
+func (ic *classSettings) validatAvailableAWSSetting(value string, availableValues []string) bool {
+	for i := range availableValues {
+		if value == availableValues[i] {
+			return true
+		}
+	}
+	return false
+}
+
+func (ic *classSettings) validateAWSSetting(value string, availableValues []string) bool {
+	for i := range availableValues {
+		if value == availableValues[i] {
+			return true
+		}
+	}
+	return false
+}
+
+func (ic *classSettings) getStringProperty(name, defaultValue string) string {
+	if ic.cfg == nil {
+		// we would receive a nil-config on cross-class requests, such as Explore{}
+		return defaultValue
+	}
+
+	value, ok := ic.cfg.ClassByModuleName("generative-aws")[name]
+	if ok {
+		asString, ok := value.(string)
+		if ok {
+			return asString
+		}
+	}
+	return defaultValue
+}
+
+func (ic *classSettings) getFloatProperty(name string, defaultValue *float64) *float64 {
+	if ic.cfg == nil {
+		// we would receive a nil-config on cross-class requests, such as Explore{}
+		return defaultValue
+	}
+
+	val, ok := ic.cfg.ClassByModuleName("generative-aws")[name]
+	if ok {
+		asFloat, ok := val.(float64)
+		if ok {
+			return &asFloat
+		}
+		asNumber, ok := val.(json.Number)
+		if ok {
+			asFloat, _ := asNumber.Float64()
+			return &asFloat
+		}
+		asInt, ok := val.(int)
+		if ok {
+			asFloat := float64(asInt)
+			return &asFloat
+		}
+	}
+
+	return defaultValue
+}
+
+func (ic *classSettings) getIntProperty(name string, defaultValue *int) *int {
+	if ic.cfg == nil {
+		// we would receive a nil-config on cross-class requests, such as Explore{}
+		return defaultValue
+	}
+
+	val, ok := ic.cfg.ClassByModuleName("generative-cohere")[name]
+	if ok {
+		asInt, ok := val.(int)
+		if ok {
+			return &asInt
+		}
+		asFloat, ok := val.(float64)
+		if ok {
+			asInt := int(asFloat)
+			return &asInt
+		}
+		asNumber, ok := val.(json.Number)
+		if ok {
+			asFloat, _ := asNumber.Float64()
+			asInt := int(asFloat)
+			return &asInt
+		}
+		wrongVal := -1
+		return &wrongVal
```

Comment:
`switch val.(type)` might work nicely here

Reply:
I will keep that in my during my next iteration of fixes for this module, for now I will leave it as it.